docs/guides/key-manager/upgrade-lsp6.md

```diff
@@ -0,0 +1,228 @@
+
+## Step 1 - Set up the constants and imports
+
+Create a javascript file and add the following imports on the top of the file.
+Create the following constants:
+
+- `privateKey`: private key of a controller address, **MUST** have [**CHANGEOWNER**](../../standards/universal-profile/lsp6-key-manager.md#permissions) permission.
+- `keyManagerAddress`: address of the current LSP6 Key Manager.
+- `universalProfileAddress`: address of your Universal Profile.
+
+```js
+import LSP6KeyManager from '@lukso/lsp-smart-contracts/artifacts/LSP6KeyManager.json' assert { type: 'json' };
+import Web3 from 'web3';
+const web3 = new Web3('https://rpc.l14.lukso.network');
+
+const privateKey = '0x...';
+const universalProfileAddress = '0x...';
+const keyManagerAddress = '0x...';
+```
+
+## Step 2 - Initialize the controller account
+
+In order to send any transaction on the blockchain you need an account, in or case that account MUST have [**CHANGEOWNER**](../../standards/universal-profile/lsp6-key-manager.md#permissions) permission on the Universal Profile that will have its LSP6 Key Manager upgraded.
+
+```js
+const account = web3.eth.accounts.privateKeyToAccount(privateKey);
+web3.eth.accounts.wallet.add(account);
+```
+
+## Step 3 - Initialize the old LSP6 Key Manager
+
+In order to transfer ownership of your Universal Profile, you need to initialize your current LSP6 Key Manager.
+
+```js
+const oldKeyManager = new web3.eth.Contract(
+  LSP6KeyManager.abi,
+  keyManagerAddress,
+);
+```
+
+## Step 4 - Deploy the new LSP6 Key Manager
+
+Deploy a new LSP6 Key Manager with the latest updates.
+
+```js
+const NewKeyManager = new web3.eth.Contract(LSP6KeyManager.abi);
+const newKeyManager = await NewKeyManager.deploy({
+  data: LSP6KeyManager.bytecode,
+  arguments: [universalProfileAddress],
+}).send({
+  from: account.address,
+  gas: 3000000,
+  gasPrice: '1000000000',
+});
+
+console.log(newKeyManager._address);
+```
+
+## Step 5 - Upgrade your Key Manager
+
+### Step 5.1 - Transfer Ownership to your new Key Manager
+
+Create a payload for the [`transferOwnership(address)`](../../standards/smart-contracts/lsp14-ownable-2-step.md#transferownership) function and shift the ownership of your Universal Profile from your current LSP6 Key Manager.
+
+```js
+const transferOwnershipPayload =
+  web3.eth.abi.encodeFunctionSignature('transferOwnership(address)') +
+  web3.eth.abi.encodeParameter('address', newKeyManager._address).substring(2);
+
+await oldKeyManager.methods['execute(bytes)'](transferOwnershipPayload).send({
+  from: account.address,
+  gas: 1000000,
+  gasPrice: '1000000000',
+});
+```
+
+### Step 5.2 - Accept Ownership from your new Key Manager
+
+Create a payload for the [`claimOwnership()`](../../standards/smart-contracts/lsp14-ownable-2-step.md#acceptownership) function and take the ownership of your Universal Profile from your new LSP6 Key Manager.
+
+```js
+const claimOwnershipPayload =
+  web3.eth.abi.encodeFunctionSignature('claimOwnership()');
```

Comment:
```suggestion
const claimOwnershipPayload =  web3.eth.abi.encodeFunctionSignature('claimOwnership()');
```

You can add //prettier-ignore above the codeblock.

It is ok to not put prettier on these examples as otherwise it will split super simple things into 1000 lines...
Having it in one line for the tutorial is fine